src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp

```diff
@@ -155,21 +204,30 @@ static constexpr size_t vec_len_f32_neon = vec_len_neon / sizeof(float);
         _mm_storeu_si128(reinterpret_cast<__m128i *>(addr), bf16_o);
     }
 
-    inline __m256 mm256_uni_loadu_ps(ov::float16* a) {
-        auto vec_f16 = _mm_loadu_si128(reinterpret_cast<__m128i*>(a));
-        auto o = _mm256_cvtph_ps(vec_f16);
-        return o;
-    }
-    inline __m256 mm256_uni_loadu_tail_ps(const ov::float16* a, const size_t count) {
-        ov::float16 tmp_values[8] = {0};
-        std::memcpy(tmp_values, a, count * sizeof(ov::float16));
-        return mm256_uni_loadu_ps(tmp_values);
-    }
     inline void mm256_uni_storeu_ps(ov::float16* a,  __m256 v) {
         __m128i vec_f16 = _mm256_cvtps_ph(v, 0);
         _mm_storeu_si128(reinterpret_cast<__m128i *>(a), vec_f16);
     }
 
+    // store __m256 to addr
+    inline void mm256_uni_storeu_tail_ps(float *addr, __m256 v, size_t count) {
+        const auto mask = get_mask(count);
+        return _mm256_maskstore_ps(addr, mask, v);
+    }
+
+    inline void mm256_uni_storeu_tail_ps(ov::bfloat16 *addr, __m256 v, size_t count) {
```

Comment:
Sohuld't we disable bf16 template specialization of Attn kernel for avx2 at all? It will allow to avoid implimentation of such functions.

Reply:
> Sohuld't we disable bf16 template specialization of Attn kernel for avx2 at all? It will allow to avoid implimentation of such functions.

Yes. You are right. This function is redundant and should not be invoked. Remove